rero_ils/modules/serializers/response.py

```diff
@@ -22,12 +22,50 @@
 """
 from __future__ import absolute_import, print_function
 
+from copy import deepcopy
 from datetime import datetime
 
 from flask import current_app
 from invenio_records_rest.serializers.response import add_link_header
 
 
+def search_responsify(serializer, mimetype):
+    """Create a Records-REST search result response serializer.
+
+    .. note::
+        Why rewrite a custom 'search_responsify' function instead of using
+        function from ``invenio_records_rest.serializers.response`` ?
+
+        The Invenio core function create a single serializer object when the
+        application is loaded. This object if reused for any requests. But
+        in our case, sometimes, we use a ``CachedDataSerializerMixin``. If the
+        cache is never reset, any changes on a resource already in the cache
+        were never applied during application instance life.
+        To solve this problem, in our custom function, we use a new
+        serializer for each call. In this way, the cache is always empty at the
+        beginning of any serialization.
+
+    :param serializer: Serializer instance.
+    :param mimetype: MIME type of response.
+    :returns: Function that generates a record HTTP response.
+    """
+    def view(pid_fetcher, search_result, code=200, headers=None, links=None,
+             item_links_factory=None):
+        copy_serializer = deepcopy(serializer)
```

Comment:
A comment is welcome to explain why the deepcopy is important.

Reply:
this comment is present into the function docstring under `..note` section